nipio/backend.py

```diff
@@ -239,8 +248,17 @@ def handle_unknown(self, qtype: str, qname: str) -> None:
         _write('LOG', f'Unknown type: {qtype}, domain: {qname}')
         _write('END')
 
-    def handle_blacklisted(self, ip_address: str) -> None:
-        _write('LOG', f'Blacklisted: {ip_address}')
+    def is_permitted(self, ip_address: IPv4Address) -> bool:
```

Comment:
Perhaps ought to be named _is_permitted, prefixed with an underscore?

Reply:
I wasn’t sure about this as none of the `handle_*` methods have a leading `_`. 

Personally I think it should be treated as part of the public API. If I was someone who was extending this class, it is a possible extension point that I might want to override.

Reply:
That is not a strong opinion though. Happy to change it if you prefer.

Reply:
This should be `_is_permitted` - it is an implementation detail that is closed to this class.